### PR TITLE
Bug fixes and optimisation

### PR DESCRIPTION
**Pull request by jarrodek**
- renders array type query parameters correctly
- produces valid `value` when data model generated from changes (fixes an issue when the URL wasn’t set without modification of any query / uri parameters)
- added support for boolean values
- added support for date format
- replaced arc icons library with Anypoint icons

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -61,14 +61,10 @@
       <h1>The raml-request-parameters-editor element</h1>
       <div main>
         <template is="dom-if" if="[[isMethod]]" restamp="true">
-          <raml-request-parameters-editor
-            query-parameters="[[selected.queryParameters]]"
-            uri-parameters="[[selected.allUriParameters]]"
-            url="[[selected.absoluteUri]]"
-            value="{{url}}"></raml-request-parameters-editor>
+          <raml-request-parameters-editor query-parameters="[[selected.queryParameters]]" uri-parameters="[[selected.allUriParameters]]" url="[[selected.absoluteUri]]" value="{{url}}"></raml-request-parameters-editor>
```
Comment (nazartokar) [inline demo/index.html:64]: We can use `<raml-request-parameters-editor ...params />` here due to we don't have children for the component. Besides that, all parameters better to have in new lines.

Try using `npm run lint` to see issues.

File: demo/index.html [hunk 2]
```diff
@@ -61,14 +61,10 @@
...
             <div class="url-value">
               <p>Produced URL value is:</p>
-              <anypoint-text-field value="{{url}}" label="Request URL"></anypoint-text-field>
+              <raml-request-url-editor auto-validate required uri-parameters="[[selected.allUriParameters]]" url="[[selected.absoluteUri]]" value="{{url}}" ></raml-request-url-editor>
```
Comment (nazartokar) [inline demo/index.html:67]:
```
<raml-request-url-editor 
auto-validate required 
uri-parameters="[[selected.allUriParameters]]" 
url="[[selected.absoluteUri]]" value="{{url}}" 
/>
```

File: test/events.html [hunk 3]
```diff
@@ -52,6 +52,7 @@
       var apiUrlRaml;
 
       suiteSetup(function(done) {
+        this.timeout(120000);
```
Comment (nazartokar) [inline test/events.html:55]: Is it correct? We need to wait 120 seconds?

File: demo/index.html (hunk 1 above)
Comment (jarrodek) [reply demo/index.html:64]: It's a demo page... :) - meaning it's not a part of the element

File: demo/index.html (hunk 2 above)
Comment (jarrodek) [reply demo/index.html:67]: Not sure what you mean by that. But 1) it's still demo page 2) I ask for a review to check if the element changes anything visually (I added new element to a collection with new styles declaration). External API of the element haven't change.

File: test/events.html (hunk 3 above)
Comment (jarrodek) [reply test/events.html:55]: That is correct. IE 11 is slow and parsing RAML on this browser may take some time.
I'm not requesting a review on a element itself. This code base was already merged with the master. As per previous comment, I'm asking about visual side.

Comment (nazartokar) [reply test/events.html:55]: I see, thanks for explaining.

**Review by nazartokar — APPROVED** [review]

**Pull request merged**